### PR TITLE
[bitnami/discourse] added missing env variables

### PR DESCRIPTION
**Pull request by ncreuschling**
### Description of the change

bitnami/discourse

### Benefits

This changes allows proper configuration via env variables to support backup/restore with an external database.

### Possible drawbacks

Unknown. Probably none.

### Additional information

see also: #bitnami/charts#12592

### CODE REVIEW

**Comment by javsalgar** [comment]
Hi! Please, note this asset is a community-supported solution. This means that the Bitnami team is not actively working on new features/improvements nor providing support through GitHub Issues. Any new issue will stay open for 20 days to allow the community to contribute, after 15 days without activity the issue will be marked as stale being closed after 5 days.

The Bitnami team will review any PR that is created, feel free to create a PR if you find any issue or want to implement a new feature.

New versions and releases cadence are not going to be affected. Once a new version is released in the upstream project, the Bitnami container image will be updated to use the latest version, supporting the different branches supported by the upstream project as usual.

**Comment by ncreuschling** [comment]
Javier, you lost me with your statements here.

I do not offer an additional feature in this PR. It's a solution to this bug report (#bitnami/charts#12592). This patch simply enables a Bitnami Discourse setup to leverage features already present in Discourse by allowing two additional environment variables to be passed on. I do not believe there to be further support effort on your part required.

**Comment by jotamartos** [comment]
Hi @ncreuschling, 

To allow Discourse get those env vars during the deployment, you need to export them to make them available when running the server. You can one example here: 

https://github.com/bitnami/containers/pull/4174/files

If the configuration file does not need to be edited, you do not need to modify the libdiscourse.sh or the README.md files

**Comment by ncreuschling** [comment]
> Hi @ncreuschling,
> 
> To allow Discourse get those env vars during the deployment, you need to export them to make them available when running the server.

I added another commit.

**Comment by jotamartos** [comment]
The new changes break the build of the docker image. 

```
 => ERROR [9/9] RUN /opt/bitnami/scripts/discourse/postunpack.sh                                                                                                                                                                         0.5s
------
 > [9/9] RUN /opt/bitnami/scripts/discourse/postunpack.sh:
#14 0.400 /opt/bitnami/scripts/discourse-env.sh: line 158: DISCOURSE_DATABASE_PORT: unbound variable
------
executor failed running [/bin/bash -o pipefail -c /opt/bitnami/scripts/discourse/postunpack.sh]: exit code: 1
```

The variable is not defined and that's why it fails. Please define the variable first and then export it. You have other examples in the same file.

**Comment by ncreuschling** [comment]
I was using the wrong name for the fallback value.

File: bitnami/discourse/2/debian-11/rootfs/opt/bitnami/scripts/discourse-env.sh [hunk 1]
```diff
@@ -152,6 +154,9 @@ DISCOURSE_DATABASE_USER="${DISCOURSE_DATABASE_USER:-"${DISCOURSE_POSTGRESQL_USER
 export DISCOURSE_DATABASE_USER="${DISCOURSE_DATABASE_USER:-bn_discourse}"
 DISCOURSE_DATABASE_PASSWORD="${DISCOURSE_DATABASE_PASSWORD:-"${DISCOURSE_POSTGRESQL_PASSWORD:-}"}"
 export DISCOURSE_DATABASE_PASSWORD="${DISCOURSE_DATABASE_PASSWORD:-}"
+export DISCOURSE_DB_BACKUP_HOST="${DISCOURSE_DB_BACKUP_HOST:-$DISCOURSE_DATABASE_HOST}"
+export DISCOURSE_DB_BACKUP_PORT="${DISCOURSE_DB_BACKUP_PORT:-$DISCOURSE_DATABASE_PORT_NUMBER}"
+
```
Comment (carrodher) [inline bitnami/discourse/2/debian-11/rootfs/opt/bitnami/scripts/discourse-env.sh:159]:
```suggestion
```

**Comment by jotamartos** [comment]
Hi @ncreuschling,

I tested the new image before accepting the change but the solution is not working. Please note that those variables are probably meant to configure the db_backup_host and db_backup_port parameters in the Discourse's conf file. The port is set to 5432 by default

https://github.com/discourse/discourse/blob/f8305f53c787ad0b6cd865d56a7b8b0f77f9dcc2/config/discourse_defaults.conf#L40-L41

You will need to adapt the scripts to configure those parameters in the conf file so Discurse knows where to  access the database when generating a backup. Once updated, you can try this testing docker-compose.yml file to confirm the changes work properly

```
version: '2'
services:
  postgresql:
    image: docker.io/bitnami/postgresql:13
    expose:
      - '5433'
    volumes:
      - 'postgresql_data:/bitnami/postgresql'
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - ALLOW_EMPTY_PASSWORD=yes
      - POSTGRESQL_USERNAME=bn_discourse
      - POSTGRESQL_DATABASE=bitnami_discourse
      - POSTGRESQL_PORT_NUMBER=5433
  redis:
    image: docker.io/bitnami/redis:7.0
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'redis_data:/bitnami/redis'
  discourse:
    image: docker.io/bitnami/discourse:testing
    ports:
      - '80:3000'
    volumes:
      - 'discourse_data:/bitnami/discourse'
    depends_on:
      - postgresql
      - redis
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - ALLOW_EMPTY_PASSWORD=yes
      - DISCOURSE_HOST=www.example.com
      - DISCOURSE_DATABASE_HOST=postgresql
      - DISCOURSE_DATABASE_PORT_NUMBER=5433
      - POSTGRESQL_PORT_NUMBER=5433
      - DISCOURSE_DATABASE_USER=bn_discourse
      - DISCOURSE_DATABASE_NAME=bitnami_discourse
      - DISCOURSE_REDIS_HOST=redis
      - DISCOURSE_REDIS_PORT_NUMBER=6379
      - POSTGRESQL_CLIENT_POSTGRES_USER=postgres
      - POSTGRESQL_CLIENT_CREATE_DATABASE_NAME=bitnami_discourse
      - POSTGRESQL_CLIENT_CREATE_DATABASE_EXTENSIONS=hstore,pg_trgm
  sidekiq:
    image: docker.io/bitnami/discourse:testing
    depends_on:
      - discourse
    volumes:
      - 'sidekiq_data:/bitnami/discourse'
    command: /opt/bitnami/scripts/discourse-sidekiq/run.sh
    environment:
      # ALLOW_EMPTY_PASSWORD is recommended only for development.
      - ALLOW_EMPTY_PASSWORD=yes
      - DISCOURSE_HOST=www.example.com
      - DISCOURSE_DATABASE_HOST=postgresql
      - DISCOURSE_DATABASE_PORT_NUMBER=5433
      - POSTGRESQL_PORT_NUMBER=5433
      - DISCOURSE_DATABASE_USER=bn_discourse
      - DISCOURSE_DATABASE_NAME=bitnami_discourse
      - DISCOURSE_REDIS_HOST=redis
      - DISCOURSE_REDIS_PORT_NUMBER=6379
volumes:
  postgresql_data:
    driver: local
  redis_data:
    driver: local
  discourse_data:
    driver: local
  sidekiq_data:
    driver: local
```

**Comment by ncreuschling** [comment]
I think I understand what you mean. I updated my commit.

**Review by jotamartos — APPROVED** [review]
Changes LGTM! I tested the new container and it configured the backup host and port for Discourse. We will work on releasing a new version of the container as soon as possible.

**Comment by jotamartos** [comment]
New version of the container was released here.

https://github.com/bitnami/containers/pull/15512/files

**Pull request merged**